ribbon-eureka/src/main/java/com/netflix/niws/loadbalancer/DiscoveryEnabledNIWSServerList.java

```diff
@@ -58,21 +60,43 @@
     boolean shouldUseOverridePort = false;
     boolean shouldUseIpAddr = false;
 
+    private final Provider<EurekaClient> eurekaClientProvider;
+
     /**
      * @deprecated use {@link #DiscoveryEnabledNIWSServerList(String)}
      * or {@link #DiscoveryEnabledNIWSServerList(IClientConfig)}
      */
     @Deprecated
     public DiscoveryEnabledNIWSServerList() {
+        this.eurekaClientProvider = new LegacyEurekaClientProvider();
     }
 
+    /**
+     * @deprecated
+     * use {@link #DiscoveryEnabledNIWSServerList(String, javax.inject.Provider)}
+     * @param vipAddresses
+     */
+    @Deprecated
     public DiscoveryEnabledNIWSServerList(String vipAddresses) {
-        IClientConfig clientConfig = DefaultClientConfigImpl.getClientConfigWithDefaultValues();
-        clientConfig.set(Keys.DeploymentContextBasedVipAddresses, vipAddresses);
-        initWithNiwsConfig(clientConfig);
+        this(vipAddresses, new LegacyEurekaClientProvider());
     }
 
+    /**
+     * @deprecated
+     * use {@link #DiscoveryEnabledNIWSServerList(com.netflix.client.config.IClientConfig, javax.inject.Provider)}
+     * @param clientConfig
+     */
+    @Deprecated
     public DiscoveryEnabledNIWSServerList(IClientConfig clientConfig) {
+        this(clientConfig, new LegacyEurekaClientProvider());
+    }
+
+    public DiscoveryEnabledNIWSServerList(String vipAddresses, Provider<EurekaClient> eurekaClientProvider) {
```

Comment:
The original usage was lazy, so I'm using a provider instead of direct injection to keep the old behaviour in case there's subtle reasons that's needed.